### PR TITLE
chore: added support for grid-*

### PR DESCRIPTION
**Pull request by anikethsaha**
Ref #756
Added normalization for grid-\*

- [X] grid-auto-flow `[ row | column ] || dense`
- [X] grid-column-gap `normal | <length-percentage>`
- [X] grid-row-gap `normal | <length-percentage>`
- [X] grid-column `<grid-line>+`
- [X] grid-row  `<grid-line>+`
- [X] grid-row-start  `<grid-line>`
- [X] grid-row-end  `<grid-line>`
- [X] grid-column-start  `<grid-line>`
- [X] grid-column-end  `<grid-line>`

### CODE REVIEW

File: packages/postcss-ordered-values/src/rules/grid.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      return `${normalizeValue.front.trim()} ${normalizeValue.back.trim()}`; // span 3
+    });
+    // returns "2 / span 3"
+    return gridValue
+      .join(' / ')
+      .trimStart()
+      .trimEnd();
+  }
+  // doing this separating if `/` is not present as while joining('/') , it will add `/` at the end
+  gridValue = gridValue.map((gridLine) => {
```
Comment (ben-eb) [inline packages/postcss-ordered-values/src/rules/grid.js]: Prefer `return x.map( ... )` to `x = x.map( ... ); return x`

File: packages/postcss-ordered-values/src/rules/grid.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        if (node === 'span') {
+          normalizeValue.front = node; // span _
+        } else {
+          normalizeValue.back = `${normalizeValue.back} ${node}`; // _ 3
+        }
+      });
+      return `${normalizeValue.front.trim()} ${normalizeValue.back.trim()}`; // span 3
+    });
+    // returns "2 / span 3"
+    return gridValue
```
Comment (ben-eb) [inline packages/postcss-ordered-values/src/rules/grid.js]: Might be nice to break this out into a utility so we can avoid the mutation on line 44 and do `return joinGridValue(gridValue.map( ... ))`

File: packages/postcss-ordered-values/src/rules/grid.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  });
+  if (shouldNormalize) {
+    return `${newValue.front.trim()} ${newValue.back.trim()}`;
+  }
+  return gridGap;
+};
+
+export const normalizeGridColumnRow = (grid) => {
+  // cant do normalization here using node, so copy it as a string
+  let gridValue = grid.toString(); // node -> string value
```
Comment (ben-eb) [inline packages/postcss-ordered-values/src/rules/grid.js]: Might be nicer as:

```js
let gridValue = grid.toString().split('/');
```

Not sure the comment adds much here?

File: packages/postcss-ordered-values/src/rules/grid.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      .trimStart()
+      .trimEnd();
+  }
+  // doing this separating if `/` is not present as while joining('/') , it will add `/` at the end
+  gridValue = gridValue.map((gridLine) => {
+    let normalizeValue = {
+      front: '',
+      back: '',
+    };
+    gridLine = gridLine.trimStart().trimEnd();
```
Comment (ben-eb) [inline packages/postcss-ordered-values/src/rules/grid.js]: This is what `[String.prototype.trim](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim)` is for 🙂

**Comment by ben-eb** [comment]
Nice work overall, have identified some minor points. 🙂

**Comment by anikethsaha** [comment]
cc @evilebottnawi

File: packages/postcss-ordered-values/src/rules/grid.js [hunk 5]
```diff
@@ -38,33 +40,30 @@ export const normalizeGridColumnRowGap = (gridGap) => {
...
-      .join(' / ')
-      .trimStart()
-      .trimEnd();
+    return joinGridValue(
+      gridValue.map((gridLine) => {
+        let normalizeValue = {
+          front: '',
+          back: '',
+        };
+        gridLine = gridLine.trimStart().trimEnd(); // '3 span ' -> '3 span'
```
Comment (ben-eb) [inline packages/postcss-ordered-values/src/rules/grid.js]:
```js
gridLine = gridLine.trim();
```

Think I mentioned this before but must have gotten lost in the machine! 🙂

**Comment by anikethsaha** [comment]
cc @ben-eb I made those changes you requested 👍

**Pull request merged**